packages/office-ui-fabric-react/src/components/SpinButton/SpinButton.test.tsx

```diff
@@ -0,0 +1,494 @@
+    expect(inputDOM.getAttribute('aria-labelledby')).to.equals(labelDOM.id);
+
+    // Assert on the label element.
+    expect(labelDOM.textContent).to.equal(exampleLabelValue);
+    expect(labelDOM.htmlFor).to.equal(inputDOM.id);
+  });
+
+  it('should increment the value in the spin button via the up button', () => {
+    const exampleLabelValue: string = 'SpinButton';
+    const exampleMinValue: number = 2;
+    const exampleMaxValue: number = 22;
+    const exampleDefaultValue: string = '12';
+
+    const renderedDOM: HTMLElement = renderIntoDocument(
+      <SpinButton
+        label={ exampleLabelValue }
+        min={ exampleMinValue }
+        max={ exampleMaxValue }
+        defaultValue={ exampleDefaultValue }
+      />
+    );
+
+    // Assert on the input element.
+    const inputDOM: HTMLInputElement = renderedDOM.getElementsByTagName('input')[0];
+    const buttonDOM: Element = renderedDOM.getElementsByClassName('ms-UpButton')[0];
+
+    expect(buttonDOM.tagName).to.equal('BUTTON');
+
+    ReactTestUtils.Simulate.mouseDown(buttonDOM,
+      {
+        type: 'mousedown',
+        clientX: 0,
+        clientY: 0
+      });
+
+    ReactTestUtils.Simulate.mouseUp(buttonDOM,
+      {
+        type: 'mouseup',
+        clientX: 0,
+        clientY: 0
+      });
+
+    expect(inputDOM.value).to.equal('13');
+    expect(inputDOM.getAttribute('aria-valuemin')).to.equal(String(exampleMinValue));
+    expect(inputDOM.getAttribute('aria-valuemax')).to.equal(String(exampleMaxValue));
+    expect(inputDOM.getAttribute('aria-valuenow')).to.equal('13');
+
+  });
+
+  it('should decrement the value in the spin button by the down button', () => {
+    const exampleLabelValue: string = 'SpinButton';
+    const exampleMinValue: number = 2;
+    const exampleMaxValue: number = 22;
+    const exampleDefaultValue: string = '12';
+
+    const renderedDOM: HTMLElement = renderIntoDocument(
+      <SpinButton
+        label={ exampleLabelValue }
+        min={ exampleMinValue }
+        max={ exampleMaxValue }
+        defaultValue={ exampleDefaultValue }
+      />
+    );
+
+    // Assert on the input element.
+    const inputDOM: HTMLInputElement = renderedDOM.getElementsByTagName('input')[0];
+    const buttonDOM: Element = renderedDOM.getElementsByClassName('ms-DownButton')[0];
+
+    expect(buttonDOM.tagName).to.equal('BUTTON');
+
+    ReactTestUtils.Simulate.mouseDown(buttonDOM,
+      {
+        type: 'mousedown',
+        clientX: 0,
+        clientY: 0
+      });
+
+    ReactTestUtils.Simulate.mouseUp(buttonDOM,
+      {
+        type: 'mouseup',
+        clientX: 0,
+        clientY: 0
+      });
+
+    expect(inputDOM.value).to.equal('11');
+    expect(inputDOM.getAttribute('aria-valuemin')).to.equal(String(exampleMinValue));
+    expect(inputDOM.getAttribute('aria-valuemax')).to.equal(String(exampleMaxValue));
+    expect(inputDOM.getAttribute('aria-valuenow')).to.equal('11');
```

Comment:
These expects are repeated throughout your tests, it might be a good idea to create some helper functions